components/form-switch/src/FormSwitch.js

```diff
@@ -0,0 +1,165 @@
+/** @jsx jsx */
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { jsx, useTheme, paint } from '@westpac/core';
+
+import { FormSwitchText, FormSwitchToggle } from './styled';
+
+// ==============================
+// Utils
+// ==============================
+
+// ==============================
+// Component
+// ==============================
+
+export const FormSwitch = ({
+	name,
+	id,
+	toggleText,
+	size,
+	block,
+	flip,
+	srOnlyText,
+	checked,
+	disabled,
+	children,
+	...props
+}) => {
+	const { formSwitch } = useTheme();
+
+	const [isChecked, setIsChecked] = useState(checked);
+	const toggle = () => setIsChecked(!isChecked);
+
+	// Common styling
+	const common = {
+		position: 'relative',
+		display: block ? 'flex' : 'inline-flex',
+		flexWrap: 'wrap',
+		verticalAlign: 'top',
+		marginRight: block ? null : formSwitch.marginRight,
+		marginBottom: formSwitch.marginBottom,
+		alignItems: 'center',
+		width: block ? '100%' : null,
+		flexDirection: flip ? 'row-reverse' : null,
+
+		':hover': {
+			cursor: 'pointer',
+		},
+	};
+
+	return (
+		<label htmlFor={id} css={common} onChange={toggle} {...props}>
+			<input
+				type="checkbox"
+				css={{
+					position: 'absolute',
+					zIndex: -1,
+					opacity: 0,
+				}}
+				name={name}
+				id={id}
+				defaultChecked={checked}
+				disabled={disabled}
+			/>
+			{children && (
```

Comment:
Do we need to test if there is children if you have it set as "required" in the propTypes? 
children: PropTypes.string.isRequired,

Reply:
Good pick up! 

I'm not sure. It can't hurt to have it checked twice I guess. Can we rely solely on prop types set to isRequired? That's a TM question

Reply:
Remember propTypes only warn, they don't stop things from breaking.